src/app/components/Loop/index.tsx

```diff
@@ -0,0 +1,425 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { AppState } from 'store/reducers';
+import './index.less';
+import {
+  getLoopOutTerms,
+  setLoop,
+  getLoopInTerms,
+  setLoopIn,
+} from 'modules/loop/actions';
+import { ButtonProps } from 'antd/lib/button';
+import { Form, Button, Icon, Radio, Tooltip, Input, Switch, message } from 'antd';
+import AmountField from 'components/AmountField';
+import InputLoopAddress from 'components/Loop/InputLoopAddress';
+import QuoteModal from './QuoteModal';
+import { Menu, Dropdown } from 'antd';
+import { ChannelWithNode } from 'modules/channels/types';
+
+interface StateProps {
+  channels: AppState['channels']['channels'];
+  url: AppState['loop']['url'];
+  lib: AppState['loop']['lib'];
+  loopTerms: AppState['loop']['loopTerms'];
+  loopQuote: AppState['loop']['loopQuote'];
+  loop: AppState['loop']['loop'];
+  error: AppState['loop']['error'];
+}
+
+interface DispatchProps {
+  getLoopOutTerms: typeof getLoopOutTerms;
+  getLoopInTerms: typeof getLoopInTerms;
+  setLoop: typeof setLoop;
+  setLoopIn: typeof setLoopIn;
+}
+
+interface State {
+  amount: string;
+  advanced: boolean;
+  isAnyValue: boolean;
+  destination: string;
+  swapRoutingFee: string;
+  swapFee: string;
+  minerFee: string;
+  prepayAmt: string;
+  channel: string;
+  conf: string;
+  htlc: boolean;
+  quoteModalIsOpen: boolean;
+  loopType: string;
```

Comment:
`loopType` should be an `enum` to avoid making mistakes with the magic string.

Reply:
added loop type to constants.ts